cansat-stm32f446/src/main.rs

```diff
@@ -144,21 +218,48 @@ mod app {
 
     #[task(local = [delay, bme])]
     fn bme_measure(ctx: bme_measure::Context) {
-        let bme = ctx.local.bme;
-        let delay = ctx.local.delay;
-        let measurements = match bme.measure(delay) {
-            Ok(m) => m,
-            Err(e) => {
-                defmt::error!("Could not read bme280 measurements: {}", Debug2Format(&e));
-                return;
-            }
+        #[cfg(feature = "bme")]
+        {
+            let bme = ctx.local.bme;
+            let delay = ctx.local.delay;
+            let measurements = match bme.measure(delay) {
+                Ok(m) => m,
+                Err(e) => {
+                    defmt::error!("Could not read bme280 measurements: {}", Debug2Format(&e));
+                    return;
+                }
+            };
+            let altitude = cansat::calculate_altitude(measurements.pressure);
+            defmt::info!("Altitude = {} meters above sea level", altitude);
+            defmt::info!("Relative Humidity = {}%", measurements.humidity);
+            defmt::info!("Temperature = {} deg C", measurements.temperature);
+            defmt::info!("Pressure = {} pascals", measurements.pressure);
+
+            bme_measure::spawn_after(5.secs()).unwrap();
+        }
+    }
+    #[task(local = [controller, filename])]
+    fn sdmmc_log(ctx: sdmmc_log::Context, log_string: String<11>) {
+        let controller = ctx.local.controller;
+        let filename = ctx.local.filename;
+        let mut volume = match controller.get_volume(embedded_sdmmc::VolumeIdx(0)) {
```

Comment:
I guess we don't really want to get volume and open the root dir everytime we log something. You can leave it as is for now, but try to think what can we move from here to init.